x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/results/StreamingChatCompletionResults.java

```diff
@@ -82,11 +83,7 @@ public record Result(String delta) implements ChunkedToXContent {
 
         @Override
         public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params params) {
-            return Iterators.concat(
-                ChunkedToXContentHelper.startObject(),
-                ChunkedToXContentHelper.field(RESULT, delta),
-                ChunkedToXContentHelper.endObject()
-            );
+            return ChunkedToXContentHelper.object(chunk((b, p) -> b.field(RESULT, delta)));
```

Comment:
This is a fine cleanup, but the whole thing seems just broken. We're dealing with one string here, there's nothing to chunk?